docs/how_to_guides/how_to_use_a_property_model.rst

```diff
@@ -55,16 +53,115 @@ users to model the chemical and physical properties of simple systems without th
     m.fs.state_block[0].dens_mass_phase['Liq']
     m.fs.state_block[0].conc_mass_phase_comp['Liq', 'NaCl']
     m.fs.state_block[0].flow_vol_phase['Liq']
-    m.fs.state_block[0].molality_comp['NaCl']
+    m.fs.state_block[0].molality_phase_comp['Liq', 'NaCl']
     m.fs.state_block[0].visc_d_phase['Liq']
-    m.fs.state_block[0].diffus_phase['Liq']
+    m.fs.state_block[0].diffus_phase_comp['Liq', 'NaCl']
     m.fs.state_block[0].enth_mass_phase['Liq']
-    m.fs.state_block[0].pressure_osm
+    m.fs.state_block[0].pressure_osm_phase['Liq']
 
     # Create the solver object.
     solver = get_solver()
 
     # Solve the model and display the output.
-    results = solver.solve(m, tee=False)
-    #m.fs.state_block[0].display()
-
+    solver.solve(m, tee=False)
+    m.fs.state_block[0].display()
+
+The following output is displayed:
+
+.. testoutput::
+
+   Block fs.state_block[0]
+
+     Variables:
+       flow_mass_phase_comp : Mass flow rate
+           Size=2, Index=fs.state_block[0].flow_mass_phase_comp_index, Units=kg/s
+           Key             : Lower : Value : Upper : Fixed : Stale : Domain
+            ('Liq', 'H2O') :   0.0 : 0.965 :  None :  True :  True : NonNegativeReals
+           ('Liq', 'NaCl') :   0.0 : 0.035 :  None :  True :  True : NonNegativeReals
+       temperature : State temperature
+           Size=1, Index=None, Units=K
+           Key  : Lower  : Value  : Upper  : Fixed : Stale : Domain
+           None : 273.15 : 298.15 : 373.15 :  True :  True : NonNegativeReals
+       pressure : State pressure
+           Size=1, Index=None, Units=Pa
+           Key  : Lower   : Value     : Upper      : Fixed : Stale : Domain
+           None : 10000.0 : 5000000.0 : 50000000.0 :  True :  True : NonNegativeReals
+       dens_mass_phase : Mass density
+           Size=1, Index=fs.properties.phase_list, Units=kg/m**3
+           Key : Lower : Value   : Upper  : Fixed : Stale : Domain
+           Liq : 500.0 : 1021.46 : 2000.0 : False : False :  Reals
+       mass_frac_phase_comp : Mass fraction
+           Size=2, Index=fs.state_block[0].mass_frac_phase_comp_index
+           Key             : Lower : Value : Upper : Fixed : Stale : Domain
+            ('Liq', 'H2O') :   0.0 : 0.965 :  None : False : False :  Reals
+           ('Liq', 'NaCl') :   0.0 : 0.035 :  None : False : False :  Reals
```

Comment:
As I understand, the expected output defined here will be compared verbatim (i.e. as a single multiline string), which would mean that the tests will fail if it the generated output differs from it by even a single decimal digit.

I remember having some discussions about the pros and cons of testing human-readable output, and as I remember the conclusion was that it leads to very brittle tests without any significant upside.

I'm not sure exactly how the `...` wildcard works, but perhaps we should limit the expected output specification to high-level information that we know will always be present (for this particular model), and use a wildcard for anything depending on actual numerical results (the validity of which is tested elsewhere).

Reply:
(I've just realized that the more "digits-heavy" sections are not shown in the highlighted section, but are below it in the same `testoutput::` block)

Reply:
The ellipsis `...` option basically matches any output. So for the output in this example, I can test certain lines and ignore other lines using the ellipsis.

Reply:
This checks the first 12 lines of the output. The remaining lines are ignored in the test. Note that this only displays the first 12 lines in the documentation too.

```
.. testoutput::

   Block fs.state_block[0]

     Variables:
       flow_mass_phase_comp : Mass flow rate
           Size=2, Index=fs.state_block[0].flow_mass_phase_comp_index, Units=kg/s
           Key             : Lower : Value : Upper : Fixed : Stale : Domain
            ('Liq', 'H2O') :   0.0 : 0.965 :  None :  True :  True : NonNegativeReals
           ('Liq', 'NaCl') :   0.0 : 0.035 :  None :  True :  True : NonNegativeReals
       temperature : State temperature
           Size=1, Index=None, Units=K
           Key  : Lower  : Value  : Upper  : Fixed : Stale : Domain
           None : 273.15 : 298.15 : 373.15 :  True :  True : NonNegativeReals
       ...
```

Reply:
I think testing the output is okay, as long as the values do not come from some optimization solver.

In this case, I would be comfortable modifying the test to just check the output block in your comment here. The `Variables` in the first 12 lines are all fixed, so we're not likely to get burned by otherwise insignificant changes to the results.

Reply:
I agree, I think this solution (showing the first N lines, or any portion of the output that's not likely to change) sounds like a good choice.